src/hotkey.rs

```diff
@@ -0,0 +1,269 @@
+    key: KeyCompare,
+}
+
+/// Something that can be compared with a keyboard key.
+#[derive(Debug, Clone, PartialEq)]
+pub enum KeyCompare {
+    Code(KeyCode),
+    Text(&'static str),
+}
+
+impl HotKey {
+    /// Create a new hotkey.
+    ///
+    /// The first argument describes the keyboard modifiers. This can be `None`,
+    /// or an instance of either [`SysMods`], or [`RawMods`]. [`SysMods`] unify the
+    /// 'Command' key on macOS with the 'Ctrl' key on other platforms.
+    ///
+    /// The second argument describes the non-modifier key. This can be either
+    /// a `&'static str` or a [`KeyCode`]. If it is a `&str`, it will be compared
+    /// against the [`unmodified text`] for a given key event; if it is a [`KeyCode`]
+    /// it will be compared against the event's key code.
+    ///
+    /// In general, [`KeyCode`] should be preferred for non-printing keys, like
+    /// the arrows or backspace.
+    ///
+    /// # Examples
+    /// ```
+    /// use druid::{HotKey, KeyEvent, RawMods, SysMods, KeyCode};
+    ///
+    /// let select_all = HotKey::new(SysMods::Cmd, "a");
+    /// let esc = HotKey::new(None, KeyCode::Escape);
+    /// let macos_fullscreen = HotKey::new(RawMods::CtrlMeta, "f");
+    /// ```
+    ///
+    /// [`KeyCode`]: enum.KeyCode.html
+    /// [`SysMods`]: enum.SysMods.html
+    /// [`RawMods`]: enum.RawMods.html
+    /// [`unmodified text`]: struct.KeyEvent.html#method.unmod_text
+    pub fn new(mods: impl Into<Option<RawMods>>, key: impl Into<KeyCompare>) -> Self {
+        HotKey {
+            mods: mods.into().unwrap_or(RawMods::None),
+            key: key.into(),
+        }
+        .warn_if_needed()
+    }
+
+    //TODO: figure out if we need to be normalizing case or something? This requires
+    //correctly documenting the expected behaviour of `unmod_text`.
+    fn warn_if_needed(self) -> Self {
+        if let KeyCompare::Text(s) = self.key {
+            let km: KeyModifiers = self.mods.into();
+            if km.shift && s.chars().any(|c| c.is_uppercase()) {
+                eprintln!(
+                    "warning: HotKey {:?} includes shift, but text is lowercase. \
+                     Text is matched literally; this may cause problems.",
+                    &self
+                );
+            }
+        }
+        self
+    }
+}
+
+/// A platform-agnostic representation of keyboard modifiers, for command handling.
+///
+/// This does one thing: it allows specifying hotkeys that use the Command key
+/// on macOS, but use the Ctrl key on other platforms.
+#[derive(Debug, Clone, Copy)]
+pub enum SysMods {
+    None,
+    /// Command on macOS, and Ctrl on windows/linux
+    Cmd,
+    /// Command + Alt on macOS, Ctrl + Alt on windows/linux
+    AltCmd,
+    /// Command + Shift on macOS, Ctrl + Shift on windows/linux
+    CmdShift,
+    /// Command + Alt + Shift on macOS, Ctrl + Alt + Shift on windows/linux
+    AltCmdShift,
+}
+
+//TODO: should something like this just _replace_ keymodifiers?
```

Comment:
This is a pretty deep question. There are potentially more than four modifier (this isn't even exotic if you distinguish left and right shift).